src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -695,14 +722,37 @@ public void ensureStartingOnNewLine()
             submit("\n");
       }
    }
+   private class Hyperlink
+   {
+      public Hyperlink(String url, String params) {
+         url_ = url;
+         params_ = params;
+      }
+
+      public String url_;
+      public String params_;
+   }
    private class ClassRange
    {
-      public ClassRange(int pos, String className, String text)
+      public ClassRange(int pos, String className, String text, Hyperlink hyperlink)
       {
          clazz  = className;
          start = pos;
          length = text.length();
+         hyperlink_ = hyperlink;
          element = Document.get().createSpanElement();
+         if (hyperlink_ != null) 
+         {
+            Event.sinkEvents(element, Event.ONCLICK);
+            Event.setEventListener(element, event -> {
+               // TODO: call the RemoteServer.consoleFollowHyperlink([url], [params]) method
```

Comment:
I don't know how to call `RemoteServer.consoleFollowHyperlink([url], [params])`. 

Probably something to `@Inject` somehow ?

Reply:
Right, you'd need to inject the appropriate "*ServerOperations" class, either in the class constructor, or via an explicit `RStudioGinjector.injectMembers(this)` call in the constructor, with an appropriate injected `initialize()` method defined. For example:

https://github.com/rstudio/rstudio/blob/a8c2fb38729956dd10b6784e611db1bd29ddf58f/src/gwt/src/org/rstudio/core/client/command/ApplicationCommandManager.java#L137-L147

https://github.com/rstudio/rstudio/blob/a8c2fb38729956dd10b6784e611db1bd29ddf58f/src/gwt/src/org/rstudio/core/client/command/ApplicationCommandManager.java#L47

Reply:
Thanks. I'll try to go the `initialize()` route

Reply:
That seems to work, although the class already had an `@Inject` constructor, so perhaps I'll move that there, but I suppose I would have to make some sort of fake `ConsoleServerOperations` for the `new VirtualConsole()` code in tests ?

Reply:
I think that's true. I'm not sure what the cleanest way to mock that in our tests would be...

Reply:
So then maybe it's easier if it's extracted to a simpler `VirtualConsoleOperations` interface.